libs/ledger-live-common/src/families/ethereum/synchronisation.ts

```diff
@@ -75,7 +81,7 @@ export const getAccountShape: GetAccountShape = async (
     currentBlockP,
     balanceP,
   ]);
-  const blockHeight = currentBlock.height.toNumber();
+  const blockHeight = currentBlock?.height.toNumber();
```

Comment:
Why is that optional chaining necessary ? 
From what I can see, `async getCurrentBlock(): Promise<Block>` always returns a height as BigNumber if it doesn't fail (which would break at the `Promise.all` level)

Reply:
I think it's because in `fetchCurrentBlock` we use `throttle`.  In line 589 `return f()` is of type `Promise<Block> | undefined`

Reply:
<img width="628" alt="Screenshot 2023-03-29 at 12 39 08" src="https://user-images.githubusercontent.com/23080211/228509087-7ffbf860-1295-48b5-a4fb-2c520e0ad8f2.png">

Reply:
Alright 👌